Source/WTF/wtf/SortedArrayMap.h

```diff
@@ -230,51 +228,47 @@ template<typename ArrayType> template<typename KeyArgument> inline bool SortedAr
     return iterator != std::end(m_array) && *iterator == *parsedKey;
 }
 
-constexpr int strcmpConstExpr(const char* a, const char* b)
+constexpr int compareSpansConstExpr(std::span<const char> a, std::span<const char> b)
 {
-    while (*a == *b && *a && *b) {
-        ++a;
-        ++b;
-    }
-    return *a == *b ? 0 : *a < *b ? -1 : 1;
+    auto commonLength = std::min(a.size(), b.size());
+    size_t i = 0;
+    while (i < commonLength && a[i] == b[i])
+        ++i;
+    auto aCharacter = i < a.size() ? a[i] : 0;
+    auto bCharacter = i < b.size() ? b[i] : 0;
```

Comment:
```suggestion
    if (a.size() != b.size())
        return a.size() < b.size() ? -1 : 1;
    auto aCharacter = a[i];
    auto bCharacter = b[i];
```

Reply:
The suggested code doesn't seem to match the original behavior?
```
if (a.size() != b.size())
        return a.size() < b.size() ? -1 : 1;
```

What was the point of finding the first character mismatch index in the above loop to then ignore it and then just compare the size. I think we're only supposed to compare the sizes if ALL the characters matched in the above loop.

Reply:
Yes I coded it wrong, but we ideally don't want to treat the character past the end of the string as a 0 even though the old code did that. When the strings are null terminated we are guaranteed that no characters are 0. But when they are not we want to correctly handle null characters.